### DISCUSSION TITLE
2FA link sent is http, only provides blank page or 404 not found

### DISCUSSION THREAD

**Discussion by erbmur** [authelia/authelia#5262] [Q&A] [answered]
Hello,

I have setup authelia to work behind haproxy running on pfsense.  One factor authentication works fine, but when tyring to access a 2FA site, the sign up email sends an http link rather then https.  Following this link just brings me to a blank page, or 404 not found.

Any suggestions?

**Comment by james-d-elliott** [comment, marked as answer]
https://www.authelia.com/integration/proxies/introduction/#required-headers

**Reply by james-d-elliott** [reply]
Sorry wrong link, edited.

**Comment by erbmur** [comment]
I have followed this as best I can, the 2FA link I recieve for device activation is still an http link, when followed it is just a blank white page.

**Reply by james-d-elliott** [reply]
The X-Forwarded-Proto header sent to authelia is indicating it's `http` not `https`. This needs to be fixed.

**Comment by erbmur** [comment]
I have setup X-Forward-Proto as required according to your documentation..or I think I have anyway.  Is there somewhere else I would need to check as well?  This is haproxy running within pfsense if that makes a difference to anything?
![image](https://user-images.githubusercontent.com/8417776/232745226-63a44253-efec-432c-b6ce-9af3cd52afd0.png)

**Reply by james-d-elliott** [reply]
Please use a container like `traefik/whoami` to debug the headers. Also how is the client connecting to HAProxy? Are they going through another proxy that connects to HAProxy via HTTP?

**Reply by erbmur** [reply]
No other proxy.  HAProxy is the only one I have setup.  I will take a look at whoami and see what turns up.

**Reply by james-d-elliott** [reply]
You don't have Cloudflare?  You could hard code it to `https`, but it's a bad idea.

**Reply by erbmur** [reply]
ooohhh, didn't evern thing about cloudflare.  Yes, I have cloudflare as well.  Probably something I need to change there too?

**Reply by erbmur** [reply]
Just an FYI, hard coding it to https also still causes the link sent to he http

**Reply by james-d-elliott** [reply]
You're probably going to have to look into what part of the process is removing it. The whoami container will probably show the same result if it's configured the same as authelia.

**Reply by schklom** [reply]
Hi from reddit @erbmur :)
The whoami container will confirm if it is a header problem. If they do not look like the ones on https://www.authelia.com/integration/proxies/introduction/#required-headers, then you will need to fix this on Cloudflare / HAProxy.

**Reply by schklom** [reply]
@erbmur For some strange reason, you added the ACL `protected-frontends` to that header.
In the Authelia doc, that ACL is used on protected domains other than Authelia's domain.
In other words, Authelia does not trigger the `protected-frontends` ACL, and therefore never receives the header.

For all X-*** headers, I cannot think of any benefit to restrict them with ACLs. Leave the ACL part blank for all common useful headers, and try again.

Try again, and check if all the X-*** headers are passed with correct values to `whoami`.

**Reply by erbmur** [reply]
You absolute legend! I had this `acl protected-frontends hdr(host) -m reg -i ^(?i)(jenkins|nextcloud|phpmyadmin)\.example\.com` and thought that was why `protected-frontends` was required with with X-*** headers.  Removeing it has fixed it and 2FA is now working as expected.  Thank you very much for your help and time!

**Reply by schklom** [reply]
No worries, glad it works now :)
